### ISSUE TITLE
jsonrpc.rs Error("data did not match any variant of untagged enum ResponseData")

### ISSUE THREAD

**Issue by avocadochicken** [onbjerg/ethers-flashbots#15]
I simulated some random tx from mempool before they were include in a block.
The line `client.inner().simulate_bundle(&bundle).await;` throwed following errors

tx `0x28c338e677518eb9a3b0ce150a54a787464b1e0ee6407ca9a2cfca3e7f1da08f` reverted and some log messages (from SushiSwap Router) "Fail with error 'TransferHelper: TRANSFER_FROM_FAILED'" are inside `value` field.

```
RelayError(
    ResponseSerdeJson {
        err: Error("data did not match any variant of untagged enum ResponseData", line: 1, column: 1249),
        text: "{\"jsonrpc\":\"2.0\",\"id\":20,\"result\":{\"bundleGasPrice\":\"1500000000\",\"bundleHash\":\"0xac94fedbf37d864f006c12945e7a50fe9e712ec25a5da833666337bca3fcecf1\",\"coinbaseDiff\":\"86709000000000\",\"ethSentToCoinbase\":\"0\",\"gasFees\":\"86709000000000\",\"results\":[{\"coinbaseDiff\":\"86709000000000\",\"error\":\"execution reverted\",\"ethSentToCoinbase\":\"0\",\"fromAddress\":\"0xb97dC3cd69a7Daf67cF3f11A9CB8BE368Bc46C6f\",\"gasFees\":\"86709000000000\",\"gasPrice\":\"1500000000\",\"gasUsed\":57806,\"revert\":\"\\u0008\\ufffdy\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000 \\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000$TransferHelper: TRANSFER_FROM_FAILED\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\",\"toAddress\":\"0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F\",\"txHash\":\"0x28c338e677518eb9a3b0ce150a54a787464b1e0ee6407ca9a2cfca3e7f1da08f\"}],\"stateBlockNumber\":13125885,\"totalGasUsed\":57806}}\n",
    },
)
```

tx `0x05261313ae0b76dc336dac8e63d66a3b348fca5d36a77418a5a94713d0bd8aa0` reverted (ouf of gas).

```
RelayError(
    ResponseSerdeJson {
        err: Error("data did not match any variant of untagged enum ResponseData", line: 1, column: 647),
        text: "{\"jsonrpc\":\"2.0\",\"id\":257,\"result\":{\"bundleGasPrice\":\"1500000000\",\"bundleHash\":\"0x9cf150b16c37243d6c3a5b31113e47f1bbe20f8cb3fee40c43263092644269bd\",\"coinbaseDiff\":\"310987500000000\",\"ethSentToCoinbase\":\"0\",\"gasFees\":\"310987500000000\",\"results\":[{\"coinbaseDiff\":\"310987500000000\",\"error\":\"execution reverted\",\"ethSentToCoinbase\":\"0\",\"fromAddress\":\"0x3438ea7671A46A493672595C0E2FB23E4C3aaA66\",\"gasFees\":\"310987500000000\",\"gasPrice\":\"1500000000\",\"gasUsed\":207325,\"toAddress\":\"0x1A2a1c938CE3eC39b6D47113c7955bAa9DD454F2\",\"txHash\":\"0x05261313ae0b76dc336dac8e63d66a3b348fca5d36a77418a5a94713d0bd8aa0\"}],\"stateBlockNumber\":13125898,\"totalGasUsed\":207325}}\n",
    },
)
```

tx `0x27757cbf58e8eedd881a0d2f4b99a340d689269e74d3bf93690b3f1f35f8546e` succeeded, but log messages are inside `value` field. https://etherscan.io/tx/0x27757cbf58e8eedd881a0d2f4b99a340d689269e74d3bf93690b3f1f35f8546e#eventlog

```
RelayError(
    ResponseSerdeJson {
        err: Error("data did not match any variant of untagged enum ResponseData", line: 1, column: 951),
        text: "{\"jsonrpc\":\"2.0\",\"id\":662,\"result\":{\"bundleGasPrice\":\"2050000000\",\"bundleHash\":\"0xb14a4c5fba855329ba679d512717e538927b7b21bf67504b3015ad5fbd18af6f\",\"coinbaseDiff\":\"386078550000000\",\"ethSentToCoinbase\":\"0\",\"gasFees\":\"386078550000000\",\"results\":[{\"coinbaseDiff\":\"386078550000000\",\"ethSentToCoinbase\":\"0\",\"fromAddress\":\"0x3D78aeBfEB21286B38E8E158bE77d2990715651a\",\"gasFees\":\"386078550000000\",\"gasPrice\":\"2050000000\",\"gasUsed\":188331,\"toAddress\":\"0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D\",\"txHash\":\"0x27757cbf58e8eedd881a0d2f4b99a340d689269e74d3bf93690b3f1f35f8546e\",\"value\":\"0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000002037c9eaac64e4519eb0b3c90000000000000000000000000000000000000000000000002fd89477d0ab3cab000000000000000000000000000000000000000000000000000000028e3825cc\"}],\"stateBlockNumber\":13125900,\"totalGasUsed\":188331}}\n",
    },
)
```

Looks like `ethers-flashbots/src/jsonrpc.rs` could see some tweaking.

**Comment by avocadochicken** [comment]
I just want to state that removing 

```
/// The value sent in this transaction.
#[serde(deserialize_with = "deserialize_u256")]
pub value: U256,
```

in `struct SimulatedTransaction` in `bundle.rs` would resolve this issue. But this might not be the best solution as we might want to return the data in the `value` field? The data is not a U256.

**Comment by onbjerg** [comment]
Oh that is weird, I assumed `value` was the amount of Eth sent? Do you have any idea what it might be?

So to fix the first issue, we just need to make `value` an `Option<T>`, but I'm unsure what `T` should be if `value` is not the amount of eth sent in the transaction.

**Comment by avocadochicken** [comment]
It more looks like char arrays. It is log / revert data. But some data is missing. I haven't put much work into it yet. This issue can wait for a few more days... :smile:

**Comment by onbjerg** [comment]
Just asked in the Flashbots Discord, it seems to be the return value of the called function, so I'll have to adjust that.

**Comment by onbjerg** [comment]
Should be fixed in 0.5.0 now according to the tests :) Let me know
